src/ObjectUtils/CoffFile.cpp

```diff
@@ -86,59 +93,213 @@ ErrorMessageOr<void> CoffFileImpl::Initialize() {
   return ErrorMessage(std::move(error));
 }
 
-static void FillDebugSymbolsFromDWARF(llvm::DWARFContext* dwarf_context,
-                                      ModuleSymbols* module_symbols) {
+std::optional<uint64_t> CoffFileImpl::GetVirtualAddressOfSymbolSection(
+    const llvm::object::SymbolRef& symbol_ref) {
+  // Symbols in the COFF symbol table contain the offset relative to the start of the section. To
+  // compute the symbol's address relative to the start of the object file (RVA), we need to find
+  // the virtual address of the section.
+  llvm::object::COFFSymbolRef coff_symbol_ref = object_file_->getCOFFSymbol(symbol_ref);
+  int32_t section_index = coff_symbol_ref.getSectionNumber();
+  llvm::Expected<const llvm::object::coff_section*> section =
+      object_file_->getSection(section_index);
+  if (!section) {
+    return std::nullopt;
+  }
+  return section.get()->VirtualAddress;
+}
+
+uint64_t kUnknownSymbolSize = std::numeric_limits<uint64_t>::max();
+
+std::optional<SymbolInfo> CoffFileImpl::CreateSymbolInfo(
+    const llvm::object::SymbolRef& symbol_ref) {
+  llvm::Expected<llvm::object::SymbolRef::Type> type = symbol_ref.getType();
+  ORBIT_CHECK(type);
+  ORBIT_CHECK(type.get() == llvm::object::SymbolRef::ST_Function);
+
+  llvm::Expected<uint32_t> flags = symbol_ref.getFlags();
+  if (!flags || (flags.get() & llvm::object::BasicSymbolRef::SF_Undefined) != 0) {
+    return std::nullopt;
+  }
+
+  // The symbol's "Value" is the offset in the section.
+  llvm::Expected<uint64_t> value = symbol_ref.getValue();
```

Comment:
Right, so the problem here is that I need to create a variable for the `llvm::Expected`. This is because I can only call `value.get()` after having checked if it contains something `if (!value)`, and I need to do this on the same variable. So what I want here is the variable to correspond to `symbol_ref.getValue()` precisely, regardless of the meaning of its content. I hope it makes sense.

Reply:
Actually, on second thought, I think that my reasoning was that this field in the symbol table is indeed called "value" in the PE specification.